### DISCUSSION TITLE
Use typescript-mongodb plugin with TypeGraphQL

### DISCUSSION THREAD

**Discussion by Merlin04** [dotansimha/graphql-code-generator#5393] [Q&A] [answered]
On my GraphQL API I use TypeGraphQL and MongoDB. I want to use `graphql-code-generator` to generate MongoDB types but I'm not sure how to do that with TypeGraphQL.

The documentation says to add `DIRECTIVES` to your `typeDefs`:
```ts
const schema = makeExecutableSchema({
  typeDefs: [
    DIRECTIVES,
    // the rest of your GraphQL types
  ],
  resolvers,
});
```

But in TypeGraphQL the schema is declared without a typeDefs field:
```ts
const schema = await buildSchema({
    resolvers
});
```

Is there a way to use the `typescript-mongodb` plugin with TypeGraphQL?

Also, what exactly is adding `DIRECTIVES` doing? Does `graphql-code-generator` run your API and get the results somehow?

**Comment by dotansimha** [comment, marked as answer]
Hi @Merlin04 ! 
I'll start with a side note: the `typescript-mongodb` was an experiment we did a long time ago, aiming to use the GraphQL (API) layer as a place to declare the DB models. This was an idea, turned into an experiment. 
My personal take is that you shouldn't mix your DB models and your APIs, since in most cases, these are different and have differences that are not minor. 

Regarding your question: Since this codegen plugin is using custom directives, you need to add it to your runtime, since otherwise you'll get an error. These directives are declarative only, and they don't have any runtime effect, it's just for making the schema valid and loadable. The Codegen plugin for mongodb uses these directives to detect which types should get a DB model and how. 

I'm not highly familiar with TypeGraphQL, but there is a logical issue with this integration you are trying to do - GraphQL codegen requires you to input a schema, and TypeGraphQL is the one that creates it. Can you please elaborate what are you trying to achieve, and how do you define your schema?
